pandas/_libs/lib.pyx

```diff
@@ -1740,6 +1754,36 @@ cpdef bint is_datetime_array(ndarray values):
     return validator.validate(values)
 
 
+cdef class DatetimeTZValidator(TemporalValidator):
+    cdef bint is_value_typed(self, object value) except -1:
+        return PyDateTime_Check(value) and (value.tzinfo is not None)
+
+    cdef inline bint is_valid_null(self, object value) except -1:
+        return is_null_datetime64(value)
+
+
+cpdef bint is_datetimetz_array(ndarray values):
+    cdef:
+        DatetimeTZValidator validator = DatetimeTZValidator(len(values),
+                                                            skipna=True)
+    return validator.validate(values)
+
+
+cdef class DatetimeTZNaValidator(TemporalValidator):
```

Comment:
you don't need this at all

Reply:
Yes I realize no this is redundant the DatetimeTZValidator works fine with missing values (and we don't need to differentiate between datetimetz and datetimetz with na either).